packages/flutter/test/widgets/app_test.dart

```diff
@@ -683,6 +683,91 @@ void main() {
     expect(copySpy.invoked, isTrue);
     expect(pasteSpy.invoked, isTrue);
   }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS, TargetPlatform.macOS }));
+
+  group('Android Predictive Back', () {
+    final List<bool> frameworkHandlesBacks = <bool>[];
+    setUp(() {
+      // Initialize to false. Because this uses a static boolean internally, it
+      // is not reset between tests or calls to pumpWidget. Explicitly setting
+      // it to false before each test makes them behave deterministically.
+      SystemNavigator.setFrameworkHandlesBack(false);
+      frameworkHandlesBacks.clear();
+      TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger
+        .setMockMethodCallHandler(SystemChannels.platform, (MethodCall methodCall) async {
+          if (methodCall.method == 'SystemNavigator.setFrameworkHandlesBack') {
+            expect(methodCall.arguments, isA<bool>());
+            frameworkHandlesBacks.add(methodCall.arguments as bool);
+          }
+          return;
+        });
+    });
+
+    tearDown(() {
+      TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger
+          .setMockMethodCallHandler(SystemChannels.platform, null);
+      SystemNavigator.setFrameworkHandlesBack(true);
```

Comment:
maybe add comment when setting it to true?

Reply:
Your comment makes me realize that this setting/resetting of this isn't necessary anymore.  It used to be tracked in a static bool that would keep state between tests, but now it's not.  I'll remove it here and elsewhere.